pvlib/ivtools/mismatch.py

```diff
@@ -0,0 +1,188 @@
+
+            resistance_series : numeric
+                series resistance :math:`R_{s}` [ohms]
+
+            resistance_shunt : numeric
+                shunt resistance :math:`R_{sh}` [ohms]
+
+            nNsVth : numeric
+                product of thermal voltage :math:`V_{th}` [V], diode
+                ideality factor :math:`n`, and number of series cells
+                :math:`N_{s}` [V]
+
+    num_pts : int
+        Number of points to compute for each IV curve.
+
+    breakdown_voltage : float
+        Vertical asymptote to use left of the y-axis. Any voltages that
+        are smaller than ``breakdown_voltage`` will be replaced by it.
+
+    Returns
+    -------
+    tuple
+        currents : np.ndarray
+            A 1D array of current values. Has shape (``num_pts``,).
+
+        voltages : np.ndarray
+            A 2D array of voltage values, where each row corresponds to
+            a single IV curve. Has shape (:math:`n`, ``num_pts``), where
+            :math:`n` is the number of IV curves passed in.
+
+    Notes
+    -----
+    This function assumes a simplified reverse bias model. When using
+    :func:`pvlib.singlediode.bishop88_v_from_i`, ``breakdown_factor`` is
+    left at the default value, which excludes the reverse bias term from
+    the model. Instead, any returned voltages that are less than
+    ``breakdown_voltage`` are replaced by it, yielding a vertical line
+    at ``breakdown_voltage``.
+
+    """
+
+    # in case params is a list containing scalars, add a dimension
+    if len(np.shape(params)) == 1:
+        params = params[np.newaxis,:]
+
+    # get range of currents from 0 to max_isc
+    max_isc = np.max(pvlib.singlediode.bishop88_i_from_v(0.0, *params.T,
+              method='newton'))
+    currents = np.linspace(0, max_isc, num=num_pts, endpoint=True)
+
+    # prepare inputs for bishop88
+    bishop_inputs = np.array([[currents[idx]]*len(params) for idx in
+                    range(num_pts)])
+    # each row of bishop_inputs contains n copies of a single current
+    # value, where n is the number of curves being added together
+    # there is a row for each current value
+
+    # get voltages for each curve
+    # (note: expecting to vectorize for both the inputted currents and
+    # the inputted curve parameters)
+    # transpose result so each row contains voltages for a single curve
+    voltages = pvlib.singlediode.bishop88_v_from_i(bishop_inputs, *params.T,
+               method='newton').T
+
+    # any voltages in array that are smaller than breakdown_voltage are
+    # clipped to be breakdown_voltage
+    voltages = np.clip(voltages, a_min=breakdown_voltage, a_max=None)
+
+    return currents, voltages
+
+
+def combine_curves(currents, voltages):
+    """
+    Combines IV curves in series.
+
+    Parameters
+    ----------
+    currents : array-like
+        A 1D array-like object. Its first element must be zero, and it
+        should be increasing.
+
```

Comment:
Could we perhaps reverse this array order? Its more natural to order current from Isc to 0, I think.

Reply:
Sure thing. Just pushed this change.